hack/generate-yamls.sh

```diff
@@ -76,6 +76,12 @@ else
   LABEL_YAML_CMD=(cat)
 fi
 
+# Turn on reconcileExternalGateway if it is configured.
+if (( RECONCILE_GATEWAY )); then
+  echo "Turning on reconcileExternalGateway"
+  echo "  reconcileExternalGateway: \"true\"" >> config/config-istio.yaml
```

Comment:
Please don't modify versioned files on the fly. This introduces noise, hidden changes, breakages (e.g., when you call this script twice with the option enabled) and potential unwanted commits.

Reply:
Done.
I added the logic in `e2e-common.sh`. Please take a look.